### ISSUE TITLE
Duplicates

### ISSUE THREAD

**Issue by matt-matt1** [mavris/MMLanScan#50]
results:
(one cell contains:)
192.168.1.25 / 255.255.255.0
(another cell contains:)
192.168.1.25 mac F0:24...(abbreviated)

Why is this repeated? (One has subnet-mask, one has mac address)

**Comment by AwaisFayyaz** [comment]
I am also getting repeat problem. one entry has mac address, other don't have. printing the console output here
```
lanScanDidFindNewDevice: Optional(<MMDevice: 0x28097b100>)
device.brand: Optional("")
device.hostname: awaiss-iphone
device.ipAddress: 192.168.10.9


lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c740>)
device.brand: Optional("")
device.hostname: awaiss-iphone
device.ipAddress: 192.168.10.9
device.macAddress: 02:00:00:00:00:00


lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c640>)
device.brand: Optional("")
device.hostname: awaiss-mac-mini
device.ipAddress: 192.168.10.8
device.macAddress: 02:00:00:00:00:00


lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c340>)
device.brand: Optional("")
device.hostname: roguepixel
device.ipAddress: 192.168.10.5
device.macAddress: 02:00:00:00:00:00


lanScanDidFindNewDevice: Optional(<MMDevice: 0x2809669c0>)
device.brand: Optional("")
device.hostname: myrouter.home
device.ipAddress: 192.168.10.1
device.macAddress: 02:00:00:00:00:00


lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c540>)
device.brand: Optional("")
device.hostname: galaxy-j7-2016
device.ipAddress: 192.168.10.7
device.macAddress: 02:00:00:00:00:00


lanScanDidFinishScanning with status: MMLanScannerStatus(rawValue: 0)
```

**Comment by sachinELEVEN** [comment]
> I am also getting repeat problem. one entry has mac address, other don't have. printing the console output here
> 
> ```
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x28097b100>)
> 
> device.brand: Optional("")
> 
> device.hostname: awaiss-iphone
> 
> device.ipAddress: 192.168.10.9
> 
> 
> 
> 
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c740>)
> 
> device.brand: Optional("")
> 
> device.hostname: awaiss-iphone
> 
> device.ipAddress: 192.168.10.9
> 
> device.macAddress: 02:00:00:00:00:00
> 
> 
> 
> 
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c640>)
> 
> device.brand: Optional("")
> 
> device.hostname: awaiss-mac-mini
> 
> device.ipAddress: 192.168.10.8
> 
> device.macAddress: 02:00:00:00:00:00
> 
> 
> 
> 
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c340>)
> 
> device.brand: Optional("")
> 
> device.hostname: roguepixel
> 
> device.ipAddress: 192.168.10.5
> 
> device.macAddress: 02:00:00:00:00:00
> 
> 
> 
> 
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x2809669c0>)
> 
> device.brand: Optional("")
> 
> device.hostname: myrouter.home
> 
> device.ipAddress: 192.168.10.1
> 
> device.macAddress: 02:00:00:00:00:00
> 
> 
> 
> 
> 
> lanScanDidFindNewDevice: Optional(<MMDevice: 0x28095c540>)
> 
> device.brand: Optional("")
> 
> device.hostname: galaxy-j7-2016
> 
> device.ipAddress: 192.168.10.7
> 
> device.macAddress: 02:00:00:00:00:00
> 
> 
> 
> 
> 
> lanScanDidFinishScanning with status: MMLanScannerStatus(rawValue: 0)
> 
> ```

From iOS 11 onwards Apple devices do not provide you with the correct MAC address of any physical device. Further in iOS 14 Apple devices do not expose their real MAC but generate a random MAC address, this ensures improved privacy of users.

**Comment by AwaisFayyaz** [comment]
@sachinELEVEN is this documented anywhere? as i have to provide this link to the client.

**Comment by sachinELEVEN** [comment]
> @sachinELEVEN is this documented anywhere? as i have to provide this link to the client.

I don’t think this issue is mentioned anywhere on Apple’s documentation but there are plenty of articles mentioning this, eg [](https://networktoolbox.de/networktoolbox-ios-11/)

I think you can still get the MAC of the devices if you run your app on simulator, but it will not work on actual iOS device.

**Comment by sachinELEVEN** [comment]
> @sachinELEVEN is this documented anywhere? as i have to provide this link to the client.

I don’t think this issue is mentioned anywhere on Apple’s documentation but there are plenty of articles mentioning this, eg https://networktoolbox.de/networktoolbox-ios-11/
 [](https://networktoolbox.de/networktoolbox-ios-11/)

I think you can still get the MAC of the devices if you run your app on simulator, but it will not work on actual iOS device.

Here is something you can try if you know the MAC of the device

**Comment by sachinELEVEN** [comment]
> @sachinELEVEN is this documented anywhere? as i have to provide this link to the client.

I don’t think this issue is mentioned anywhere on Apple’s documentation but there are plenty of articles mentioning this, eg https://networktoolbox.de/networktoolbox-ios-11/
  
I think you can still get the MAC of the devices if you run your app on simulator, but it will not work on actual iOS device.
